storage/innobase/log/log0log.cc

```diff
@@ -1165,6 +1180,18 @@ void log_write_up_to(lsn_t lsn, bool durable,
   }
 }
 
+void log_t::writer_update() noexcept
+{
+  ut_ad(latch_have_wr());
+  ut_ad(resize_in_progress() != 1);
+  writer=
+#ifdef HAVE_PMEM
+    is_mmap() ? log_writer<false,true> :
+#endif
+    resize_in_progress() ? log_writer<true> : log_writer<false>;
```

Comment:
We may actually have `resize_in_progress() == 1` here: https://buildbot.mariadb.org/#/builders/234/builds/35089/steps/11/logs/stdio
The reason is simply that the caller did not assign `log_sys.resize_lsn` yet; it would after this call returns. I think that we must simply remove this assertion.

We do have an assertion on `resize_in_progress() > 1` in `log_t::write_buf()`, which would be inlined in `log_writer<false>` and `log_writer<true>`.